app/mqtt_writer.py

```diff
@@ -0,0 +1,37 @@
+"""MQTT Writer module.
+
+This module provides a client class for publishing messages to MQTT brokers.
+
+"""
+
+import logging
+
+import paho.mqtt.client as mqtt
+
+from app.message import ErrorMessage
+from app.mqtt_client import MqttClient
+
+
+class MqttWriter(MqttClient):
+    _client: mqtt.Client
+
+    def __init__(
+        self,
+        port: int,
+        host: str,
+        client: mqtt.Client = None,
+    ) -> None:
+        super().__init__(host, port, client)
+        self._client.on_publish = self._on_publish()
+
+    def publish(self, topic, payload):
```

Comment:
similar to what I commented above about the reader, the topic, is somehting we should be able to know upon instantiation.
However, because we would like to allow to send error messages to different MQTT suffixes, what I think we should let the client indicate, is what's the suffix. So, rather than receiving the entire `topic`, we should rename this as `suffix: string`

Reply:
Also, can we add a type to the `payload` property?

Reply:
What would `type` be used for? Is it similar to `category` or additional?

I can see a bit of confusion here ... I have two `publish` methods:
- one in `ErrorHandler` which is opinionated about the payload structure
- and one here in `MQTTWriter` which is trying to be quite generic (but still calls `ErrorMessage.write` to do the serialisation).
 
My thinking was, if `MQTTWriter` is used in future to publish messages to other non-error-related topics, it would need to accept the entire topic string rather than append to the prefix configured for errors - so I allowed the full topic to be specified here. But maybe this is premature and it's ok for `MQTTWriter` to assume that it's always publishing error messages? Or should it be completely unopinionated about topic/payload (in which case ErrorMessage should serialise the payload before passing it in)?

Reply:
I think it's a valid question. I'm not strongly opinionated here and I see your point now.

The goal of this project though is not to be a bidirectional tool..l the goal is to send commands to the edge. Based on that, I feel the mqtt writer doesn't have to be generic.

I would even consider to avoid that abstraction and just encapsulate that inside the error handler. That's not needed though, having separation of concerns is also good. I'll let you to decide there. Having that in mind, I agree that the underlying layer if you keep it, should be generic. Acting more like a driver. The wrapper layer, the error handler class is the one that would be in charge to have deeper control and expose a dedicated interface

Reply:
ok thanks - I've kept MQTTWriter and moved the serialisation back to ErrorHandler, making this a completely generic driver for publishing to any MQTT topic - so I've left the topic as fully determined by the caller. I think it's worth keeping to encapsulate specifying the `qos` and handling the callback.